Adding unit tests for SeLion-Grid

ysecaran opened the pull request:
- Added tests for FileExtractor , ConfigParser , AuthenticationHelper,
  SauceConfigReader and ArtifactDetails.

ysecaran commented:
@mach6 @sundaramrajendran @fakrudeen78  i am done adding unit tests to SeLion-Grid.

Here is the travis run: https://travis-ci.org/ysecaran/SeLion/jobs/63917703

krmahadevan commented on server/src/test/java/com/paypal/selion/utils/AuthenticationHelperTest.java:
```diff
@@ -0,0 +1,71 @@
+        assertFalse(isValidUser);
+    }
+
+    @AfterClass
```
please add `alwaysRun=true` so that this runs always.

ysecaran commented on server/src/test/java/com/paypal/selion/utils/AuthenticationHelperTest.java:
```diff
@@ -0,0 +1,71 @@
+    @Test(dependsOnMethods = "testPasswordChange")
+    public void authenticateNewPassword() {
+        boolean val = AuthenticationHelper.authenticate("admin", "dummy");
+        System.out.println(val);
```
This will be removed while addressing review comments

ysecaran commented on server/src/test/java/com/paypal/selion/utils/ConfigParserTest.java:
```diff
@@ -0,0 +1,37 @@
+import org.powermock.modules.testng.PowerMockTestCase;
+import org.testng.annotations.Test;
+
+public class ConfigParserTest extends PowerMockTestCase {
```
The power mock references will be removed when addressing the review comment

ysecaran commented:
@krmahadevan i addressed the review comments and the travis build looks good too.

mach6 commented on server/src/test/resources/config/sauceConfig.json:
```diff
@@ -0,0 +1,4 @@
+{
```
Where in the current tests is this file used? `AuthenticationHelperTest`?

ysecaran replied:
@mach6 thanks for pointing that out. it was checked in as part of new files. will remove it

ysecaran commented:
@mach6 removed it.

mach6 commented:
@ysecaran the `.tar.bz2` and `.zip` files are corrupt when they are checked out on a Windows machine. I need you to add entries to the `.gitattributes` file to treat them as binary with this PR.

otherwise, 

:+1: LGTM .. Holding off on merging this until you update the PR and until I tag the code for the M4 changes.

Commit:
Adding unit tests for SeLion-Grid

- Added tests for FileExtractor , ConfigParser , AuthenticationHelper,
  SauceConfigReader, FileDownloader and ArtifactDetails.
- Updated .gitattributes file

ysecaran commented:
@mach6 i updated the pull request.

mach6 commented:
@ysecaran thank you.. this looks good. I will merge after M4 is "released".

mach6 commented:
this is merged. thank you
